packages/pds/src/services/local/index.ts

```diff
@@ -140,11 +141,41 @@ export class LocalService {
       handle: res.handle,
       displayName: record?.displayName,
       avatar: record?.avatar
-        ? this.getImageUrl(did, record.avatar.ref.toString(), 'avatar')
+        ? this.getImageUrl('avatar', did, record.avatar.ref.toString())
         : undefined,
     }
   }
 
+  async formatAndInsertPostsInFeed(
+    feed: FeedViewPost[],
+    posts: RecordDescript<PostRecord>[],
+  ): Promise<FeedViewPost[]> {
+    if (posts.length === 0) {
+      return feed
+    }
+    const lastTime = feed.at(-1)?.post.indexedAt ?? new Date(0).toISOString()
+    const inFeed = posts.filter((p) => p.indexedAt > lastTime)
+    const newestToOldest = inFeed.reverse()
+    const maybedFormatted = await Promise.all(
+      newestToOldest.map((p) => this.getPost(p)),
+    )
+    const formatted = maybedFormatted.filter((p) => p !== null) as PostView[]
+    for (const post of formatted) {
+      let inserted = false
+      for (let i = 0; i < feed.length; i++) {
+        if (feed[i].post.indexedAt < post.indexedAt) {
+          feed.splice(i, 0, { post })
+          inserted = true
+          break
```

Comment:
Spidey senses tingled a little seeing a loop over `feed` that also mutates `feed`, but this should be fine since it causes a break.

Reply:
cleaned it up a bit with a `findIndex` instead 👌